addons/mass_mailing/tests/common.py

```diff
@@ -104,14 +104,15 @@ def assertMailTraces(self, recipients_info, mailing, records,
                 'MailTrace: email %s (recipient %s, state: %s, record: %s): found %s records (1 expected)' % (email, partner, state, record, len(recipient_trace))
             )
             self.assertTrue(bool(recipient_trace.mail_mail_id_int))
+            if 'failure_type' in recipient_info or state in ('ignored', 'exception', 'canceled', 'bounced'):
```

Comment:
Are you sure it's a "or" here?

Reply:
Well actually we should always have both to have complete tests. The "or" was there mainly when updating tests to detect incomplete tests (missing failure type). As those are tools used mainly by me, well, will probably keep it like that.